src/CMakeLists.txt

```diff
@@ -81,8 +81,24 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL GNU)
   endif()
 endif(CMAKE_CXX_COMPILER_ID STREQUAL GNU)
 
-set(CMAKE_ASM_COMPILER  ${PROJECT_SOURCE_DIR}/src/yasm-wrapper)
-set(CMAKE_ASM_FLAGS "-f elf64")
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "ppc64le")
+  message(STATUS " we are ppc64le")
+  CHECK_C_COMPILER_FLAG("-maltivec" HAS_ALTIVEC)
+  if(HAS_ALTIVEC)
+    message(STATUS " HAS_ALTIVEC yes")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -maltivec")
+    set(CMAKE_CXX_FLAGS "${CMAKE_C_FLAGS} -maltivec")
+  endif(HAS_ALTIVEC)
+  CHECK_C_COMPILER_FLAG("-mcpu=power8" HAS_POWER8)
+  if(HAS_POWER8)
+    message(STATUS " HAS_POWER8 yes")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DHAS_POWER8")
```

Comment:
As suggested, I used src/include/config-h.in.cmake.  I renamed HAS_POWER8 to be HAVE_POWER8 to better fit with the naming convention within config-h.in.cmake.  I added #include "acconfig.h" to src/common/crc32c_ppc.c.